### PR TITLE
Displays "Export layer" when a sql has been applied and fixes bug in the options menu

### LINKED ISSUE #2820: Add "Export layer" when a query is applied in a Map
### STEPS TO REPRODUCE:
- Create a map/visualization with several layers.
- Apply a query in any of those layers.
- Click on the 'edit' link (upper right).
- Export layer doesn't appear as an option.

Ideally, we should let users export the current selected query layer as we do in table view.

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #2820 and #2898.

### CODE REVIEW

**Comment by alonsogarciapablo** [comment]
@xavijam can you PTAL? thanks!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2527/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/2528/))

File: lib/assets/javascripts/cartodb/table/header/options_menu.js [hunk 1]
```diff
@@ -39,9 +39,9 @@ cdb.admin.HeaderOptionsMenu = cdb.admin.DropdownMenu.extend({
 
   render: function() {
     var opts = this.options;
-    this.table = this.model.map.layers && this.model.map.layers.last().table;
     this.user = this.options.user;
     this.dataLayer = this.options.dataLayer;
+    this.table = this.dataLayer.table;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/header/options_menu.js]: Jum, ok!

**Comment by xavijam** [comment]
Looks good :)

File: lib/assets/javascripts/cartodb/table/header/views/options_menu.jst.ejs [hunk 2]
```diff
@@ -41,18 +41,19 @@
...
 
-      <li class="<% if (table.isSync()) { %>disabled<% } %>">
-        <a class="small georeference" href="#/georeference">
-          Georeference layer
-          <div class="progress-bar">
-            <div class="bar-2"></div>
-          </div>
-        </a>
-      </li>
+      <% if (!dataLayer.get('query')) { %>
```
Comment (javisantana) [inline lib/assets/javascripts/cartodb/table/header/views/options_menu.jst.ejs]: as a comment: we try to not access objects in rendering, it's better to send the things you need to render to the template

**Pull request merged**